R/annotate.R

```diff
@@ -0,0 +1,120 @@
+#' Annotates messy dates for uncertainty and/or approximation
+#'
+#' Some datasets have an arbitrary cut off point
+#' for start and end points, for example,
+#' but these do not refer to the real start or end dates.
+#' Though these are often coded as precise dates.
+#' This collection of functions helps annotate uncertainty and
+#' approximation to dates according to ISO2019E standards.
+#' Innacurate start or end dates can be represented by an affix
+#' indicating "on or before", if used as a prefix (e.g. ..1816-01-01),
+#' or indicating "on or after", if used as a suffix (e.g. 2016-12-31..).
+#' Approximate dates are indicated by adding a `~` to year,
+#' month, or day, to estimate values that are
+#' possibly correct (e.g. 2003~-03-03).
+#' Day, month, or year, uncertainty can be indicated by adding a `?`
+#' to a possibly dubious date (e.g. ?1916-10-10) or date
+#' component (e.g. 1916-10-10?).
+#' @param x A date vector
+#' @param level Should the annotation be on the "year", "month", "day",
+#' or month and day ("md")? Optional.
+#' If unspecified annotation will be added to before date.
+#' For month and day ("md") uncertainty or approximation
+#' sign is added before month in date (e.g. 1916-?10-10 or 1916-~10-10).
```

Comment:
If there is a default action for when left unspecified, this is not really optional. Is `level` clear enough to users? I'm not sure what else to call it, but I wonder whether this is clear enough for new users?

Reply:
@jhollway @henriquesposito shall I change 'level' to 'component'?

Reply:
Date `component` is good! Let's use that consistently throughout the package.